test/tests/functional/pbs_soft_walltime.py

```diff
@@ -0,0 +1,634 @@
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        # Sleep till just after dedicated time starts
+        time.sleep(61)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+
+        self.server.expect(JOB, {'estimated.soft_walltime':
+                                 (MATCH_RE, '00:00:65|65')}, id=jid)
+
+    def test_soft_before_prime(self):
+        """
+        Make sure that if a job's soft_walltime won't complete before
+        prime boundry, the job does not start
+        """
+        self.scheduler.set_sched_config({'backfill_prime': 'True'})
+
+        self.setup_holidays(3600, 7200)
+
+        J = Job(TEST_USER)
+        jid = self.server.submit(J)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 5400})
+        comment = 'Not Running: Job will cross into primetime'
+        self.server.expect(JOB, {'comment': comment}, id=jid)
+
+    def test_soft_backfill_prime(self):
+        """
+        Test if soft_walltime is used to see if a job can run before
+        the next prime boundry
+        """
+        self.scheduler.set_sched_config({'backfill_prime': 'True'})
+
+        self.setup_holidays(60, 3600)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+
+        J = Job(TEST_USER, {'Resource_List.walltime': 300})
+        jid = self.server.submit(J)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 5})
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+        time.sleep(61)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {'estimated.soft_walltime':
+                                 (MATCH_RE, '00:00:65|65')}, id=jid)
+
+    def test_resv_conf_soft(self):
+        """
+        Test that there is no change in the reservation behavior with
+        soft_walltime set on jobs with no hard walltime set
+        """
+
+        a = {'resources_available.ncpus': 4}
+        self.server.create_vnodes(name='vn', attrib=a, num=1, mom=self.mom)
+
+        now = int(time.time())
+        J = Job(TEST_USER, attrs={'Resource_List.ncpus': 4})
+        jid = self.server.submit(J)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 5})
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        a = {'Resource_List.ncpus': 1, 'reserve_start': now + 10,
+             'reserve_end': now + 130}
+        R = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(R)
+        self.server.log_match(rid + ';reservation deleted', max_attempts=5)
+
+    def test_resv_conf_soft_with_hard(self):
+        """
+        Test that there is no change in the reservation behavior with
+        soft_walltime set on jobs with a hard walltime set.  The soft_walltime
+        should be ignored and only the hard walltime should be used.
```

Comment:
It is not clear from the test that reservation gets deleted due to hard walltime. The behavior is exactly the same as test above where you were only setting soft walltime. May be you edit the test and set walltime to 59 and start reservation at now +60 and make sure it starts.

Reply:
The two tests are testing distinct behavior.  One is testing a job with soft_walltime and no walltime.  The soft_walltime should not be used and the job should be considered taking forever.  The other is testing a job with soft_walltime and walltime, but the walltime is longer than the reservation start time.  This one proves that the soft_walltime is ignored and the walltime is used.  The test you are suggesting is just testing existing behavior.  It should exist in TestReservations, but doesn't.

Reply:
Bhroam, I get the test cases. In first case if you do not specify the walltime will it be taking default, i.e., 5years?